File: backend/app/models/weather.py
from pydantic import BaseModel
from typing import Optional, List


class WeatherCondition(BaseModel):
    id: int
    main: str
    description: str
    icon: str


class WindInfo(BaseModel):
    speed: float
    deg: float
    gust: Optional[float] = None


class CurrentWeather(BaseModel):
    city: str
    country: str
    lat: float
    lon: float
    temp: float
    feels_like: float
    temp_min: float
    temp_max: float
    humidity: int
    pressure: int
    visibility: Optional[int] = None
    wind: WindInfo
    weather: List[WeatherCondition]
    sunrise: int
    sunset: int
    dt: int
    timezone: int


class ForecastItem(BaseModel):
    dt: int
    temp: float
    feels_like: float
    temp_min: float
    temp_max: float
    humidity: int
    pressure: int
    wind: WindInfo
    weather: List[WeatherCondition]
    pop: float  # probability of precipitation


class ForecastDay(BaseModel):
    date: str
    temp_min: float
    temp_max: float
    humidity: float
    wind_speed: float
    weather: WeatherCondition
    pop: float
    items: List[ForecastItem]


class ForecastResponse(BaseModel):
    city: str
    country: str
    lat: float
    lon: float
    forecast: List[ForecastDay]


class AIInsight(BaseModel):
    summary: str
    highlights: List[str]
    recommendation: str


class WeatherWithInsights(BaseModel):
    current: CurrentWeather
    forecast: ForecastResponse
    ai_insight: Optional[AIInsight] = None

File: frontend/src/types/weather.ts
export interface WeatherCondition {
  id: number;
  main: string;
  description: string;
  icon: string;
}

export interface WindInfo {
  speed: number;
  deg: number;
  gust?: number;
}

export interface CurrentWeather {
  city: string;
  country: string;
  lat: number;
  lon: number;
  temp: number;
  feels_like: number;
  temp_min: number;
  temp_max: number;
  humidity: number;
  pressure: number;
  visibility?: number;
  wind: WindInfo;
  weather: WeatherCondition[];
  sunrise: number;
  sunset: number;
  dt: number;
  timezone: number;
}

export interface ForecastItem {
  dt: number;
  temp: number;
  feels_like: number;
  temp_min: number;
  temp_max: number;
  humidity: number;
  pressure: number;
  wind: WindInfo;
  weather: WeatherCondition[];
  pop: number;
}

export interface ForecastDay {
  date: string;
  temp_min: number;
  temp_max: number;
  humidity: number;
  wind_speed: number;
  weather: WeatherCondition;
  pop: number;
  items: ForecastItem[];
}

export interface ForecastResponse {
  city: string;
  country: string;
  lat: number;
  lon: number;
  forecast: ForecastDay[];
}

export interface AIInsight {
  summary: string;
  highlights: string[];
  recommendation: string;
}

export interface WeatherWithInsights {
  current: CurrentWeather;
  forecast: ForecastResponse;
  ai_insight?: AIInsight;
}

export type UnitSystem = 'metric' | 'imperial' | 'standard';
